pandas/tests/indexing/test_indexing.py

```diff
@@ -1018,3 +1018,20 @@ def test_validate_indices_high():
 def test_validate_indices_empty():
     with tm.assert_raises_regex(IndexError, "indices are out"):
         validate_indices(np.array([0, 1]), 0)
+
+
+def test_modify_with_duplicate_index():
+    trange = pd.date_range(start=pd.Timestamp(year=2017, month=1, day=1),
+                           end=pd.Timestamp(year=2017, month=1, day=5))
+
+    # insert a duplicate element to the index
+    trange = trange.insert(loc=5, item=pd.Timestamp(year=2017, month=1, day=5))
+
+    df = pd.DataFrame(0, index=trange, columns=["A", "B"])
+    bool_idx = np.array([False, False, False, False, False, True])
+
+    # modify the value for the duplicate index entry
```

Comment:
this is fine as a test, but please add another case like the original issue (e.g. ``+=``)